x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -286,7 +294,16 @@ export class TaskManagerRunner implements TaskRunner {
       if (apmTrans) apmTrans.end('failure');
       performanceStopMarkingTaskAsRunning();
       this.onTaskEvent(asTaskMarkRunningEvent(this.id, asErr(error)));
-      if (error.statusCode !== VERSION_CONFLICT_STATUS) {
+      if (!SavedObjectsErrorHelpers.isConflictError(error)) {
+        if (!SavedObjectsErrorHelpers.isNotFoundError(error)) {
+          // try to release claim as an unknown failure prevented us from marking as running
+          mapErr((errReleaseClaim: Error) => {
+            this.logger.error(
+              `[Task Runner] Task ${this.instance.id} failed to release claim after failure: ${errReleaseClaim}`
+            );
+          }, await promiseResult(this.releaseClaimAndIncrementAttempts()));
```

Comment:
Do we need this `promiseResult` if `releaseClaimAndIncrementAttempts` is returning a `promiseResult`? (Sorry if this is an obvious question)

Reply:
Good catch!
Indeed unnecessary 🙏